lib/rester/utils/stub_file.rb

```diff
@@ -0,0 +1,101 @@
+module Rester
+  module Utils
+    class StubFile
+      DEFAULT_TAGS = { 'successful' => 'true' }.freeze
+
+      attr_reader :path
+
+      def initialize(path)
+        @path = path
+        @_stub = StubFile.parse(path)
+      end
+
+      private
+
+      def method_missing(meth, *args, &block)
+        if @_stub.respond_to?(meth)
+          @_stub.public_send(meth, *args, &block)
+        else
+          super
+        end
+      end
+
+      ##
+      # Class Methods
+      class << self
+        ##
+        # Parses the stub file and returns the data as a hash
+        def parse(path)
+          parse!(YAML.load_file(path))
```

Comment:
Just wondering: what are the benefits of designing the `parse` + `parse!` methods like this instead of having all of the `parse!` logic in `parse` with the top line being `stub_hash = YAML.load_file(path)`?